types/lib/model.d.ts

```diff
@@ -372,8 +372,19 @@ export type WhereAttributeHash<TAttributes = any> = {
    *      [Op.gt]: 20
    *    }
    *  }
+   *
+   * - An $attribute$ name
+   */
+  [attribute in keyof TAttributes as attribute extends string ? attribute | `$${attribute}$` : attribute]?: WhereValue<TAttributes> | WhereOptions<TAttributes>;
+} & {
+  /**
+   * Makes $nested.syntax$ valid, but does not type-check the name of the include nor the name of the include's attribute.
    */
-  [field in keyof TAttributes]?: WhereValue<TAttributes> | WhereOptions<TAttributes>;
+  // TODO [2022-05-26]: Remove this ts-ignore once we drop support for TS < 4.4
+  // TypeScript < 4.4 does not support using a Template Literal Type as a key.
+  //  note: this *must* be a ts-ignore, as it works in ts >= 4.4
+  // @ts-ignore
+  [attribute: `$${string}.${string}$`]: WhereValue<TAttributes> | WhereOptions<TAttributes>;
```

Comment:
won't this still be broken in ts <4.4?

It might be better to do something like `& { [attribute: string]: WhereValue<any>` (iirc it'll keep the other keys typed but anything else will be accepted, can't recall for certain though)

Reply:
It will still be broken in TS < 4.4, but not more than before (it's just going to ignore this addition)

Strict typings in `where` is so important IMO that it's not worth it to dumb it down for old TS versions.
I've caught a handful of broken `where` in my own code after properly typing my attributes.

If someone wants to fix this in TS < 4.4, I'd say open a follow-up PR that uses [`typesVersion`](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#version-selection-with-typesversions) to provide TS < 4.4 specific code.